cli/cliui/agent.go

```diff
@@ -121,26 +132,101 @@ func Agent(ctx context.Context, writer io.Writer, opts AgentOptions) error {
 		resourceMutex.Unlock()
 		switch agent.Status {
 		case codersdk.WorkspaceAgentConnected:
+			// NOTE(mafredri): Once we have access to the workspace agent's
+			// startup script logs, we can show them here.
+			// https://github.com/coder/coder/issues/2957
+			if agent.DelayLoginUntilReady && !opts.SkipDelayLoginUntilReady {
+				switch agent.LifecycleState {
+				case codersdk.WorkspaceAgentLifecycleCreated, codersdk.WorkspaceAgentLifecycleStarting:
+					select {
+					case <-messageAfterDone:
+						showMessage()
+					default:
+						// This state is normal, we don't want
+						// to show a message prematurely.
+					}
+				case codersdk.WorkspaceAgentLifecycleReady:
+					return nil
+				default:
```

Comment:
Yay or nay?

```suggestion
				case case codersdk.WorkspaceAgentLifecycleStartError:
					showMessage()
					return errors.New("...")
				default:
```

Reply:
`case case`?

Reply:
I think we should fail-open here. Let folks try to connect, but warn that it might not work.

Reply:
Yup, I agree. Let's make it so!

Reply:
Done, this is what it looks like now:

```
❯ coder ssh test

 > The workspace agent ran into a problem during startup.
   See troubleshooting instructions at:
   https://coder.com/docs/coder-oss/latest/templates#troubleshooting-templates

Agent startup script exited with non-zero status, use --no-wait to login anyway.
Run 'coder ssh --help' for usage.
```

And with `--no-wait` we still warn about the problem:

```
❯ coder ssh test --no-wait

 > The workspace agent ran into a problem during startup.
   See troubleshooting instructions at:
   https://coder.com/docs/coder-oss/latest/templates#troubleshooting-templates

mafredri@test ~
❯
```